src/vos/vos_io.c

```diff
@@ -394,8 +394,21 @@ akey_fetch_recx(daos_handle_t toh, const daos_epoch_range_t *epr,
 			void *csum_ptr = daos_csum_from_offset(csum,
 				(uint32_t) ((lo - recx->rx_idx) * rsize));
 
-			memcpy(csum_ptr, ent->en_csum.cs_csum,
-			       csum_nr * ent->en_csum.cs_len);
+			if (DAOS_FAIL_CHECK(DAOS_CHECKSUM_FETCH_FAIL)) {
+				void *fault_ptr;
+
+				D_ALLOC_ARRAY(fault_ptr,
+					csum_nr * ent->en_csum.cs_len);
+				memset(fault_ptr, 3,
+					csum_nr * ent->en_csum.cs_len);
+				memcpy(csum_ptr, fault_ptr,
+					csum_nr * ent->en_csum.cs_len);
+				D_FREE(fault_ptr);
```

Comment:
hmm, why allocate fault_ptr here? it seems you can do
memset(csum_ptr, random(), csum_nr * ent->en_csum.cs_len) directly?

Reply:
Thanks. Done.